site/content/integrate/slash-commands/custom/_index.md

```diff
@@ -0,0 +1,137 @@
+    - (_Optional_) Select the **Autocomplete** option to include the slash command in the command autocomplete list, displayed when typing ``/`` in an empty input box. Use it to make your command easier to discover by your teammates. You can also provide a hint listing the arguments of your command and a short description displayed in the autocomplete list.
+
+      {{<note "Note:">}}
+   [Enable integrations to override usernames](https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-usernames) must be set to `true` in `config.json` to override usernames. Enable them from **System Console > Integrations > Integration Management**, or ask your System Admin to do so. If not enabled, the username is set to `webhook`.
+
+   Similarly, [Enable integrations to override profile picture icons](https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-profile-picture-icons) must be set to `true` in `config.json` to override profile picture icons. Enable them from **System Console > Integrations > Integration Management**, or ask your System Admin to do so. If not enabled, the icon of the creator of the webhook URL is used to post messages.
+      {{</note>}}
+
+3. Select **Save**. On the next page, copy the **Token** value. This will be used in a later step.
+
+   ![image](slash_commands_token.png)
+
+4. Next, write your external application. Include a function which receives HTTP `POST` or HTTP `GET` requests from Mattermost. The request will look something like this:
+
+    ```http request
+    POST /slash-command HTTP/1.1
+    Host: example.com
+    Accept: application/json
+    Accept-Encoding: gzip
+    Authorization: Token qzgakf1nx3yt9dr4n8585ihbxy
+    Content-Length: 313
+    Content-Type: application/x-www-form-urlencoded
+    User-Agent: Mattermost-Bot/1.1
+    
+    channel_id=jux16pkewjrkfj3ehep1psxyxc&
+    channel_mentions=["saepe-5", "aut-8"]&
+    channel_mentions_ids=["r3j6sby343fpfdxcbwqg95rfsa", "ehjj46yk7ifptr5bpfb966s6mc"]&
+    channel_name=town-square&
+    command=%2Ftest&
+    response_url=http%3A%2F%2Flocalhost%3A8065%2Fhooks%2Fcommands%2Fxbrkb8p393gjpq5cawei7npije&
+    team_domain=test&
+    team_id=carya1qs77bemjup96ff538snh&
+    text=asd&
+    token=okwexkjpe7ygb8eq1ww58t483w&
+    user_id=aoa1agao6t8fmx3ikt1j9w5ybw&
+    user_mentions=["aaron.peterson", "aaron.medina"]&
+    user_mentions_ids=["q5s3b7xzgprp5eid8h66j9epsy", "czwmumrmw7dfxecww7qibkkoor"]
+    user_name=somename&
+    ```
+
+   If your integration sends back a JSON response, make sure it returns the `application/json` content-type.
+
+5. Add a configurable *MATTERMOST_TOKEN* variable to your application and set it to the **Token** value from step 7. This value will be used by your application to confirm the HTTP `POST` or `GET` request came from Mattermost.
```

Comment:
Also, I see no reason to prescribe that they use the variable name `MATTERMOST_TOKEN`, it could be any arbitrary variable anywhere on your application. You only need to make your appropriate point in the second sentence in step 5, that it used to validate the request's origin.

Reply:
Step "7" should really be step "3" since a bunch of steps were moved into bullet points and threw off the original numbering. I'll reword this step.